pkg/dmaasbackup/periodic_backup_test.go

```diff
@@ -0,0 +1,602 @@
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dmaasbackup
+
+import (
+	"context"
+	"reflect"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	core "k8s.io/client-go/testing"
+
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/mayadata-io/dmaas-operator/pkg/apis/mayadata.io/v1alpha1"
+	"github.com/mayadata-io/dmaas-operator/pkg/builder"
+	"github.com/mayadata-io/dmaas-operator/pkg/generated/clientset/versioned/fake"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	velerobuilder "github.com/vmware-tanzu/velero/pkg/builder"
+	velerofake "github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned/fake"
+	veleroinformer "github.com/vmware-tanzu/velero/pkg/generated/informers/externalversions"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryclock "k8s.io/apimachinery/pkg/util/clock"
+)
+
+func newTestScheduleSpec() velerov1api.ScheduleSpec {
+	return velerov1api.ScheduleSpec{
+		Schedule: "*/2 * * * *",
+	}
+}
+
+func veleroScheduleDetails(
+	t *testing.T,
+	scheduleTime string,
+	status v1alpha1.VeleroScheduleStatus,
+	isReserved bool,
+) v1alpha1.VeleroScheduleDetails {
+	testTime, err := time.Parse("2006-01-02 15:04:05", scheduleTime)
+	require.NoError(t, err, "Failed to parse schedule time: %v", err)
+
+	var creationTime metav1.Time
+
+	if !isReserved {
+		creationTime.Time = testTime
+	} else {
+		// for reserved veleroschedule, set status empty explicitly
+		status = ""
+	}
+
+	return v1alpha1.VeleroScheduleDetails{
+		ScheduleName:      "name-" + testTime.Format("20060102150405"),
+		CreationTimestamp: &creationTime,
+		Status:            status,
+	}
+}
+
+func compareVeleroSchedule(given, expected []v1alpha1.VeleroScheduleDetails) error {
+	sort.Sort(sort.Reverse(ScheduleByCreationTimestamp(expected)))
+	if (given == nil) != (expected == nil) {
+		return errors.New("either of schedule is empty")
+	}
+
+	if len(given) != len(expected) {
+		return errors.Errorf("Length mismatched given-len=%d expected-len=%d", len(given), len(expected))
+	}
+
+	for i := range given {
+		if !reflect.DeepEqual(given[i], expected[i]) {
+			return errors.Errorf("given schedule=%v, expected=%v", given[i], expected[i])
+		}
+	}
+
+	return nil
+}
+
+func TestProcessPeriodicConfigSchedule(t *testing.T) {
+	dbkpbuilder := func() *builder.DMaaSBackupBuilder {
+		return builder.ForDMaaSBackup("ns", "name")
```

Comment:
lets put some constants for these strings..
these constants will help us when we revisit the code later

Reply:
done